magma/backend/mlir/hardware_module.py

```diff
@@ -277,10 +277,13 @@ def __init__(self, graph: Graph, ctx):
         self._ctx = ctx
         self._visited = set()
 
+    def new_value(self, T):
```

Comment:
let's just expose `ctx` as a property rather than delegate this function up the chain